client-spark/shuffle-manager-3/src/main/java/org/apache/spark/shuffle/rss/HashBasedShuffleWriter.java

```diff
@@ -165,7 +184,11 @@ public void write(scala.collection.Iterator<Product2<K, V>> records) throws IOEx
     } else {
       write0(records);
     }
-    close();
+    if (isColumnarShuffle) {
```

Comment:
I suggest putting closeColumnarWrite in close method，and we could reuse some code at the same time.